float64[3] alpha
float64[3] beta
float64[3] gamma